src/tdastro/astro_utils/opsim.py

```diff
@@ -3,146 +3,196 @@
+        x = np.cos(dec_rad) * np.cos(ra_rad)
+        y = np.cos(dec_rad) * np.sin(ra_rad)
+        z = np.sin(dec_rad)
+        cart_coords = np.array([x, y, z]).T
+
+        # Construct the kd-tree.
+        self._kd_tree = KDTree(cart_coords)
+
+    @classmethod
+    def from_db(cls, filename, sql_query="SELECT * FROM observations", colmap=_rubin_opsim_colnames):
+        """Create an OpSim object from the data in an opsim db file.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the opsim db file.
+        sql_query : `str`
+            The SQL query to use when loading the table.
+            Default = "SELECT * FROM observations"
+        colmap : `dict`
+            A mapping of short column names to their names in the underlying table.
+            Defaults to the Rubin opsim column names.
+
+        Returns
+        -------
+        opsim : `OpSim`
+            A table with all of the pointing data.
+
+        Raise
+        -----
+        ``FileNotFoundError`` if the file does not exist.
+        ``ValueError`` if unable to load the table.
+        """
+        if not Path(filename).is_file():
+            raise FileNotFoundError(f"opsim file {filename} not found.")
+        con = sqlite3.connect(f"file:{filename}?mode=ro", uri=True)
+
+        # Read the table.
+        try:
+            opsim = pd.read_sql_query(sql_query, con)
+        except Exception:
+            raise ValueError("Opsim database read failed.") from None
+
+        # Close the connection.
+        con.close()
+
+        return OpSim(opsim, colmap=colmap)
+
+    def write_opsim_table(self, filename, tablename="observations", overwrite=False):
+        """Write out an opsim database to a given SQL table.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the opsim db file.
+        tablename : `str`
+            The table to which to write.
+            Default = "observations"
+        overwrite : `bool`
+            Overwrite the existing DB file.
+            Default = False
+
+        Raise
+        -----
+        ``FileExistsError`` if the file already exists and ``overwrite`` is ``False``.
+        """
+        if_exists = "replace" if overwrite else "fail"
+
+        con = sqlite3.connect(filename)
+        try:
+            self.table.to_sql(tablename, con, if_exists=if_exists)
+        except Exception:
+            raise ValueError("Opsim database write failed.") from None
+
+        con.close()
+
+    def range_search(self, query_ra, query_dec, fov):
+        """Return the indices of the opsim pointings that fall within the field
+        of view of the query point(s).
+
+        Parameters
+        ----------
+        query_ra : `float` or `numpy.ndarray`
+            The query right ascension (in degrees).
+        query_dec : `float` or `numpy.ndarray`
+            The query declination (in degrees).
+        fov : `float`
```

Comment:
Usually we think about fov to be a square root of area or diameter or edge length  we shouldn't define it as radius, I'd prefer to use diameter here. Also we may use LSST value as a default, diameter is 3.5 degrees